float64 distance
---
bool result
---
float64 closest_point